rfcs/2019-04-01-externally-changed-file.md

```diff
@@ -0,0 +1,53 @@
+# RFC: Acting upon an externally changed file
+
+Right now Xi only has two mechanisms to handle externally changed files:
+
+* Blocking saving, instead requiring the user to save the file elsewhere
+* Automatically reloading the file if no changes have been done
+
+Blocking saving is a bit annoying, especially if the user _does_ want to overwrite
+the file (e.g. when a git rebase has modified a file's modtime).
+
+## Summary
+
+For this to work we have to change multiple thing:
+
+- Add a new notification ('file_externally_changed')
+	- Frontends may use this to display a 'File has been changed, want to reload?' dialog
+- Add a way for frontends to force reload the file to implement the previously mentioned dialog
+- Add a way for frontends to force save the file, e.g. via a 'force' param to 'save'
+
+### Force saving
+
+There are multiple ways to tacle this:
+
+1.
+If the frontend sends 'save' with 'force: false' and the file has been changed
+externally, then Xi should send another notification back, e.g. 'save_failed'
```

Comment:
I think that probably `save` should report errors in general; I think it makes sense for save to become a `request`. Maybe this would be a good first step for this work? We would want to support the existing API fro the time being, but we could have both notification & request live alongside one another for a while.

Reply:
I'd have to check, but I think so; we separate notifications from requests early (based on the presence of the `id` field) so this should work. Let's try it out?

Reply:
Alright, will adjust the RFC in a bit! :)